common/os/pfwrapper.c

```diff
@@ -434,9 +434,14 @@ pf_ll_setup(struct _perf_cpu *cpu, pf_conf_t *conf)
 	attr.precise_ip = 1;
 	attr.exclude_guest = 1;
 	attr.sample_type = PERF_SAMPLE_TID | PERF_SAMPLE_ADDR | PERF_SAMPLE_CPU |
-		PERF_SAMPLE_WEIGHT | PERF_SAMPLE_CALLCHAIN;
+		PERF_SAMPLE_WEIGHT | PERF_SAMPLE_CALLCHAIN |
+		PERF_SAMPLE_DATA_SRC;
 	attr.disabled = 1;
 
+	/* Unlike Core PMUs, others like IBS do not support guest exclusion */
+	if (attr.type != PERF_TYPE_RAW && attr.type != PERF_TYPE_HARDWARE)
+		attr.exclude_guest = 0;
```

Comment:
The LL event is a special case that uses a "precise" event. If I am not mistaken, Intel uses PEBS for this and it happens to be an extension of the Core PMU itself (at least from the perspective of the `perf_event_open` syscall). However, that is not the case for AMD processors as they need to use the IBS Fetch or Op PMUs which are separate from the Core PMU and have their own non-ABI `type` value. `PERF_TYPE_RAW` and `PERF_TYPE_HARDWARE` are generally tied to the Core PMU. This is why on Intel, they can be used for PEBS events but not on AMD for IBS events.

Rather than moving the comment, would it be better to make it more descriptive instead?
I prefer keeping it here since this applies only to the LL event.

Reply:
Or should we add a flag in struct _plat_event_config for that instead of special type? So that's more clear for behavior difference.

Reply:
I see your point now. It does make sense to have that extra customization option by including the `exclude_guest` attribute as a part of the struct.